magellan-library/src/main/java/com/wealthfront/magellan/navigation/NavigationDelegate.kt

```diff
@@ -149,10 +150,8 @@ public open class NavigationDelegate(
     navigationPropagator.onNavigatedTo(currentNavigable)
     when (currentState) {
       is LifecycleState.Shown, is LifecycleState.Resumed -> {
-        containerView!!.addView(
-          currentNavigable.view!!,
-          direction.indexToAddView(containerView!!)
-        )
+        val templatedView = templateApplier?.onViewCreated(currentNavigable.view!!) ?: currentNavigable.view!!
```

Comment:
`navigateTo` and `navigateFrom` should return the templated view rather than the navigable's view